metals/src/main/scala/scala/meta/internal/metals/Embedded.scala

```diff
@@ -61,44 +60,59 @@ final class Embedded(
 }
 
 object Embedded {
-  def downloadSettings(
+  def fetchSettings(
       dependency: Dependency,
       scalaVersion: String
-  ): Settings =
-    new coursiersmall.Settings()
-      .withTtl(Some(Duration.Inf))
-      .withDependencies(List(dependency))
+  ): Fetch = {
+    val fetch = Fetch
+      .create()
+      .withDependencies(List(dependency): _*)
+      .addRepositories(
+        Repository.defaults().asScala.toList: _*
+      )
       .addRepositories(
         List(
-          coursiersmall.Repository.SonatypeReleases,
-          coursiersmall.Repository.SonatypeSnapshots
-        )
+          Repository.central(),
+          Repository.ivy2Local()
+        ): _*
       )
-      .withForceVersions(
+      .withDependencies(
         List(
-          new Dependency(
+          Dependency.of(
             "org.scala-lang",
             "scala-library",
             scalaVersion
           ),
-          new Dependency(
+          Dependency.of(
             "org.scala-lang",
             "scala-compiler",
             scalaVersion
           ),
-          new Dependency(
+          Dependency.of(
             "org.scala-lang",
             "scala-reflect",
             scalaVersion
           )
-        )
+        ): _*
       )
 
+    val envRepos = System
+      .getenv("COURSIER_REPOSITORIES")
+    if (envRepos.nonEmpty)
```

Comment:
it would be nicer to edit the repositories at the top and then pass them to the `Fetch` directly, like:

```scala
def foo = {
  val repositories = 
    if(envRepos.isEmpty) DefaultRepositories
    else DefaultRepositories ++ envRepos.map(MavenRepository.of)
  
  // this is a last statement
  Fetch
    (...)
    .withRepositories(repositories)
}
```